spec/lib/jyson/jyson.array.spec.js

```diff
@@ -15,41 +15,216 @@ describe('jyson.array.spec: an array in the template', () => {
+          b: [2, 3],
+          c: [4, 5, 6]
+        };
+        const json = this.templateFunction(input);
+
+        expect(json.a).to.have.ordered.members(input.a);
+        expect(json.a).to.have.lengthOf(input.a.length);
+        expect(json.b).to.have.ordered.members(input.b);
+        expect(json.b).to.have.lengthOf(input.b.length);
+        expect(json.c).to.have.ordered.members(input.c);
+        expect(json.c).to.have.lengthOf(input.c.length);
+      });
+    });
+
+    describe('complex arrays', () => {
+      it('must convert an object to json', () => {
+        const input = {
+          a: [1],
+          b: [2, 3],
+          c: [4, 5, 6]
+        };
+        input.a.test = 'hello world';
+        input.b.push('hello world');
+
+        const json = this.templateFunction(input);
+
+        expect(json.a).to.have.ordered.members([1]);
+        expect(json.a).to.have.lengthOf(input.a.length);
+        expect(json.b).to.have.ordered.members(input.b);
+        expect(json.b).to.have.lengthOf(input.b.length);
+        expect(json.c).to.have.ordered.members(input.c);
+        expect(json.c).to.have.lengthOf(input.c.length);
+      });
     });
   });
 
-  describe('complex arrays', () => {
-    it('must convert an object to "json"', () => {
-      const input = {
-        a: [1],
-        b: [2, 3],
-        c: [4, 5, 6]
-      };
-      input.a.test = 'hello world';
-      input.b.push('hello world');
-
-      const json = this.templateFunction(input);
-
-      expect(json.a).to.have.ordered.members([1]);
-      expect(json.a).to.have.lengthOf(input.a.length);
-      expect(json.b).to.have.ordered.members(input.b);
-      expect(json.b).to.have.lengthOf(input.b.length);
-      expect(json.c).to.have.ordered.members(input.c);
-      expect(json.c).to.have.lengthOf(input.c.length);
+  describe('template arrays defined by objects', () => {
+    describe('simple arrays', () => {
+      beforeEach(() =>{
+        this.templateFunction = jyson.buildTemplateFunction({
+          a: [{
+            a: 'a.$.a',
+            b: 'a.$.b',
+            c: 'a.$.c'
+          }]
+        });
+      });
+
+      describe('simple arrays', () => {
+        it('must convert an object to json', () => {
+          const input = {
+            a: [
+              {
+                a: 'a0a',
+                b: 'a0b',
+                c: 'a0c',
+              }, {
+                a: 'a1a',
+                b: 'a1b',
+                c: 'a1c',
+              }
+            ],
+          };
+          const json = this.templateFunction(input);
+          expect(json.a.length).to.equal(2);
+          expect(json.a[0].a).to.equal('a0a');
+          expect(json.a[0].b).to.equal('a0b');
+          expect(json.a[0].c).to.equal('a0c');
+          expect(json.a[1].a).to.equal('a1a');
+          expect(json.a[1].b).to.equal('a1b');
+          expect(json.a[1].c).to.equal('a1c');
+        });
+
+        it('must handel undefined values an object to json', () => {
```

Comment:
Suggest changing spec to `must convert undefined values to null`? Not sure if that's the intention here